gtk/gtk-play.c

```diff
@@ -1051,6 +1051,47 @@ create_tracks_menu (GtkPlay * play, GstPlayerMediaInfo * media_info, GType type)
   return menu;
 }
 
+static gboolean
+player_set_rate_cb (GtkRange * range, GtkScrollType scroll,
+    gdouble value, GtkPlay * play)
+{
+  gst_player_set_rate (play->player, value);
+}
+
+static void
+player_speed_clicked_cb (GtkButton * button, GtkPlay * play)
+{
+  GtkWidget *scale;
+  GtkWidget *dialog;
+  GtkWidget *content;
+  GtkWidget *box;
+  gdouble value;
+
+  dialog = gtk_dialog_new ();
+  gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (play));
+  gtk_window_set_title (GTK_WINDOW (dialog), "Playback speed control");
+
+  content = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+  box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 1);
+  gtk_box_pack_start (GTK_BOX (content), box, TRUE, TRUE, 0);
+
+  scale = gtk_scale_new_with_range (GTK_ORIENTATION_HORIZONTAL,
```

Comment:
Maybe we can just put a GtkScaleButton directly into the menu instead of creating a dialog? Or even better a GtkSpinButton directly into the menu.

I think a spin button is also better UI for this in general.


While at it, I think we should also block the seekbar position-update code while the seekbar is dragged. Otherwise if you slowly drag it around, it will jump back and forth all the time between the position it just seeked to and the position the mouse drags it to